Lib/contextlib.py

```diff
@@ -4,9 +4,10 @@
 import _collections_abc
 from collections import deque
 from functools import wraps
+import inspect
```

Comment:
This introduces a dependency I'd prefer to avoid - `inspect` pulls in a *lot* of other modules, and `contextlib` is used frequently enough that we should aim to minimise its dependency tree.